src/consensus/aft/raft.h

```diff
@@ -296,16 +296,29 @@ namespace aft
       return state->membership_state == kv::MembershipState::Retired;
     }
 
-    void set_retired_committed(ccf::SeqNo seqno) override
-    {
-      state->retirement_phase = kv::RetirementPhase::RetiredCommitted;
-      CCF_ASSERT_FMT(
-        state->retired_committed_idx == state->commit_idx,
-        "Retired "
-        "committed index {} does not match current commit index {}",
-        state->retired_committed_idx.value_or(0),
-        state->commit_idx);
-      state->retired_committed_idx = seqno;
+    void set_retired_committed(
+      ccf::SeqNo seqno, const std::vector<kv::NodeId>& node_ids) override
+    {
+      for (auto& node_id : node_ids)
+      {
+        if (id() == node_id)
+        {
+          CCF_ASSERT(
+            state->membership_state == kv::MembershipState::Retired,
+            "Node is not retired, cannot become retired committed");
+          CCF_ASSERT(
+            state->retirement_phase == kv::RetirementPhase::Completed,
+            "Node is not retired, cannot become retired committed");
+          state->retirement_phase = kv::RetirementPhase::RetiredCommitted;
+          CCF_ASSERT_FMT(
+            state->retired_committed_idx == state->commit_idx,
+            "Retired "
+            "committed index {} does not match current commit index {}",
+            state->retired_committed_idx.value_or(0),
+            state->commit_idx);
```

Comment:
```suggestion
          CCF_ASSERT_FMT(
            state->retired_committed_idx == state->commit_idx,
            "Retired "
            "committed index {} does not match current commit index {}",
            state->retired_committed_idx.value_or(0),
            state->commit_idx);
          state->retirement_phase = kv::RetirementPhase::RetiredCommitted;
```

Minor: _if_ we ever fail one of these sanity check preconditions, prefer to update neither state value.